recognition/advanced_tree_search.py

```diff
@@ -283,13 +285,16 @@ def create_config(
         cpu: int,
         lmgc_mem: float,
         lmgc_alias: Union[None, str],
+        lmgc_scorer: Union[None, rasr.FeatureScorer],
```

Comment:
What is the difference to `Optional[rasr.FeatureScorer]`? Just curious.

Reply:
My thinking was: Optional can mean "You can give it, but you do not have to, and then its None"
Here it is always given (due to the call and no default)

Reply:
`Optional[...] is a shorthand notation for Union[..., None]` but it seems this was wrong thinking from my side, so I can change it if you want.

Reply:
I agree with Willi, IIRC in the surrounding code this is `Union` but in `i6_*` repos it is mainly `Optional`... I would at least keep this per file consistent.. :)

Reply:
I hope I didn't overlook anything.